File: footprint-frontend/frontend/src/services/emissionFactorService.js
import api from "./api";

export const getActiveEmissionFactors = () => {
  return api.get("/user/emission-factors");
};
export const userCreateEmissionFactor = (data) => {
  return api.post("/user/emission-factors", data);
};
// Admin functions (for future admin pages)

export const getAllEmissionFactors = () => {
  return api.get("/admin/emission-factors");
};

export const getEmissionFactorById = (id) => {
  return api.get(`/admin/emission-factors/${id}`);
};

export const createEmissionFactor = (data) => {
  return api.post("/admin/emission-factors", data);
};

export const updateEmissionFactor = (id, data) => {
  return api.put(`/admin/emission-factors/${id}`, data);
};

export const deleteEmissionFactor = (id) => {
  return api.delete(`/admin/emission-factors/${id}`);
};

File: footprint-backend/footprint-backend/src/main/java/com/carbonfootprint/footprint_backend/service/EmissionFactorService.java
package com.carbonfootprint.footprint_backend.service;

import com.carbonfootprint.footprint_backend.dto.EmissionFactorRequest;
import com.carbonfootprint.footprint_backend.dto.EmissionFactorResponse;
import com.carbonfootprint.footprint_backend.entity.EmissionFactor;

import java.util.List;

public interface EmissionFactorService {

    // Used internally by ActivityService
    EmissionFactor getEmissionFactorById(Long id);

    // User side
    List<EmissionFactorResponse> getActiveEmissionFactors();

    // Admin side
    List<EmissionFactorResponse> getAllEmissionFactors();

    EmissionFactorResponse getEmissionFactor(Long id);

    EmissionFactorResponse createEmissionFactor(
            EmissionFactorRequest request
    );

    EmissionFactorResponse updateEmissionFactor(
            Long id,
            EmissionFactorRequest request
    );

    void deleteEmissionFactor(Long id);

}